### ISSUE TITLE
Go test race conditions on `./pkg/endpoints`

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/dashboard#73]
There is some data race (caught by `go test -race`) on `./pkg/endpoints`.

- on `TestLogWebsocket`
```
{"level":"info","msg":"Adding API for websocket"}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
==================
WARNING: DATA RACE
Read at 0x00c00047aee0 by goroutine 86:
  github.com/tektoncd/dashboard/pkg/websocket.poll()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:73 +0x1ca

Previous write at 0x00c00047aee0 by goroutine 85:
  github.com/tektoncd/dashboard/pkg/websocket.poll.func1()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:63 +0x46
  github.com/tektoncd/dashboard/vendor/github.com/gorilla/websocket.(*Conn).advanceFrame()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/gorilla/websocket/conn.go:892 +0xac3
  github.com/tektoncd/dashboard/vendor/github.com/gorilla/websocket.(*Conn).NextReader()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/gorilla/websocket/conn.go:947 +0x10a
  github.com/tektoncd/dashboard/vendor/github.com/gorilla/websocket.(*Conn).ReadMessage()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/gorilla/websocket/conn.go:1028 +0x50
  github.com/tektoncd/dashboard/pkg/websocket.readControl()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:51 +0x50

Goroutine 86 (running) created at:
  github.com/tektoncd/dashboard/pkg/websocket.WriteOnlyWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:45 +0x99
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:37 +0xa0
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket-fm()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:32 +0x70
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:288 +0x108a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch-fm()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:203 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /usr/local/go/src/net/http/server.go:2375 +0x28a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).ServeHTTP()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:303 +0x6e
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2774 +0xc4
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1878 +0x807

Goroutine 85 (running) created at:
  github.com/tektoncd/dashboard/pkg/websocket.WriteOnlyWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:44 +0x77
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:37 +0xa0
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket-fm()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:32 +0x70
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:288 +0x108a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch-fm()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:203 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /usr/local/go/src/net/http/server.go:2375 +0x28a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).ServeHTTP()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:303 +0x6e
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2774 +0xc4
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1878 +0x807
==================
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
==================
WARNING: DATA RACE
Write at 0x00c0000b11c0 by goroutine 88:
  github.com/tektoncd/dashboard/pkg/broadcaster.(*Broadcaster).Unsubscribe()
      /go/src/github.com/tektoncd/dashboard/pkg/broadcaster/broadcaster.go:123 +0xd7
  github.com/tektoncd/dashboard/pkg/websocket.readControl()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:53 +0x121

Previous read at 0x00c0000b11c0 by goroutine 74:
  github.com/tektoncd/dashboard/pkg/websocket.write()
      /go/src/github.com/tektoncd/dashboard/pkg/broadcaster/broadcaster.go:51 +0x53
  github.com/tektoncd/dashboard/pkg/websocket.WriteOnlyWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:46 +0xb1
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:37 +0xa0
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket-fm()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:32 +0x70
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:288 +0x108a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch-fm()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:203 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /usr/local/go/src/net/http/server.go:2375 +0x28a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).ServeHTTP()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:303 +0x6e
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2774 +0xc4
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1878 +0x807

Goroutine 88 (running) created at:
  github.com/tektoncd/dashboard/pkg/websocket.WriteOnlyWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/websocket/websocket.go:44 +0x77
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:37 +0xa0
  github.com/tektoncd/dashboard/pkg/endpoints.Resource.establishPipelineLogsWebsocket-fm()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket.go:32 +0x70
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:288 +0x108a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).dispatch-fm()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:203 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /usr/local/go/src/net/http/server.go:2375 +0x28a
  github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful.(*Container).ServeHTTP()
      /go/src/github.com/tektoncd/dashboard/vendor/github.com/emicklei/go-restful/container.go:303 +0x6e
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2774 +0xc4
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1878 +0x807

Goroutine 74 (running) created at:
  net/http.(*Server).Serve()
      /usr/local/go/src/net/http/server.go:2884 +0x4c4
  net/http/httptest.(*Server).goServe.func1()
      /usr/local/go/src/net/http/httptest/server.go:298 +0xac
==================
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
websocket_test.go:49: Enter TestLogWebsocket...
websocket_test.go:64: Waiting for clients to terminate...
websocket_test.go:74: Exit TestLogWebsocket
testing.go:809: race detected during execution of test
```

- on `TestPipelineRunWebsocket`

```
{"level":"info","msg":"Adding API for websocket"}
{"level":"debug","msg":"Into StartPipelineRunController"}
{"level":"info","msg":"PipelineRun Controller Started"}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
{"level":"debug","msg":"Upgrading connection to websocket..."}
Creating pipelinerun
{"level":"debug","msg":"In pipelineRunCreated"}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunCreated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:123456] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
Updating pipelinerun WebsocketPipelinerun
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunUpdated map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
Deleting pipelinerun: WebsocketPipelinerun
{"level":"debug","msg":"In pipelineRunDeleted"}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
{PipelineRunDeleted map[metadata:map[creationTimestamp:<nil> name:WebsocketPipelinerun namespace:ns1 resourceVersion:654321] spec:map[Status: params:<nil> pipelineRef:map[name:] resources:<nil> serviceAccount: trigger:map[type:]] status:map[conditions:<nil>]]}
==================
WARNING: DATA RACE
Read at 0x00000290c598 by goroutine 12:
  github.com/tektoncd/dashboard/pkg/endpoints.TestPipelineRunWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket_test.go:115 +0x569
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163

Previous write at 0x00000290c598 by goroutine 140:
  github.com/tektoncd/dashboard/pkg/endpoints.clientWebsocket.func1()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket_test.go:172 +0x3ff

Goroutine 12 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:916 +0x699
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1157 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1155 +0x523
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1072 +0x2eb
  github.com/tektoncd/dashboard/pkg/endpoints.TestMain()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/routes_test.go:82 +0xfba
  main.main()
      _testmain.go:86 +0x222

Goroutine 140 (running) created at:
  github.com/tektoncd/dashboard/pkg/endpoints.clientWebsocket()
      /go/src/github.com/tektoncd/dashboard/pkg/endpoints/websocket_test.go:143 +0x31c
==================
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
{"level":"error","msg":"websocket connection to client lost: websocket: close 1005 (no status)"}
websocket_test.go:79: Enter TestPipelineRunWebsocket...
websocket_test.go:117: Waiting for clients to terminate...
websocket_test.go:127: Exit TestPipelineRunWebsocket
testing.go:809: race detected during execution of test
```

I disabled race detection on the CI for now (to bootstrap it), but this need to be fixed (so that we can re-enable race detections :angel: )

cc @a-roberts 

See https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_dashboard/72/pull-tekton-dashboard-tests/1120985522127245312/

**Comment by a-roberts** [comment]
I'll take a look at this.

**Comment by a-roberts** [comment]
Let's see if I can assign OK now, Vincent wrote most of this code so we're going to buddy up (and if it doesn't work I get to search the repos for me saying this and retrospectively assign him)

/assign @vtereso

**Comment by a-roberts** [comment]
@vtereso produced a fix for this y'day and I've tidied things up, added more comments, merged with master and resolved an import problem. I've also added the -race check back in.

https://github.com/tektoncd/dashboard/compare/master...a-roberts:websocketRacev2?expand=1 here's the code on my branch and here's the original: https://github.com/vtereso/dashboard/tree/websocketRace.

I'm going to wait until Vincent is back online and then he can replace the contents of his branch with mine, squash commits and get the PR in so he can get the credit. Thanks @vtereso!
